securedrop_client/logic.py

```diff
@@ -419,14 +431,16 @@ def on_sync_success(self) -> None:
 
     def on_sync_failure(self, result: Exception) -> None:
         """
-        Called when syncronisation of data via the API fails after a background sync. Resume the
-        queues so that we continue to retry syncing with the server in the background.
+        Called when syncronisation of data via the API fails after a background sync. If the reason
+        a sync fails is ApiInaccessibleError then we need to log the user out for security reasons
+        and show them the login window in order to get a new token.
         """
         logger.debug('The SecureDrop server cannot be reached due to Error: {}'.format(result))
-        self.gui.update_error_status(
-            _('The SecureDrop server cannot be reached.'),
-            duration=0,
-            retry=True)
+
+        if isinstance(result, ApiInaccessibleError):
+            self.invalidate_token()
```

Comment:
very minor nit: maybe `invalidate_token_locally` or something? (just because otherwise the dev might think this method calls `Api.logout`)